taurus_pyqtgraph/datainspectormodel.py

```diff
@@ -0,0 +1,205 @@
+                        self.triggerPoint(curve, point_index, i)
+
+                    except KeyError:
+
+                        # Create the TextItem object for current curve and init
+                        # the point style
+                        self.informationClouds[i] = TextItem()
+
+                        parent = self.plot.curves[i]
+                        self.informationClouds[i].setParentItem(parent)
+
+                        self.scatter[i] = curve.scatter
+                        if curve.opts['symbol'] is None:
+                            # create the custom style of the point
+                            # if point style (symbol and size) not exist
+                            curve.setSymbol('o')
+                            curve.setSymbolSize(10)
+                            self.custom_point_style_init = True
+
+                        self.paintCloud(point_x, point_y, i)
+                        self.triggerPoint(curve, point_index, i)
+
+                else:
+                    try:
+                        # Clean the information cloud
+                        # when the line isn't near any plot point
+                        self.informationClouds[i].setText("")
+                        curve.updateItems()
+                    except KeyError:
+                        pass
+
+    def paintCloud(self, x, y, i):
+        """
+        Method use to painting the cloud with the data
+
+        :param x: the new x position of the picked point
+        :param y: the new y position of the picked point
+        :param i: the index of current curve
+        """
+        self.informationClouds[i].setPos(x, y)
+        self.informationClouds[i].setHtml("<div style='%s'> "
+                                           "<span>x=%s "
+                                           "<span>y=%0.1f</span> "
+                                           "</div>" % (self.cloudStyle,
+                                                       self.timestampToTime(x),
+                                                       y))
+
+    def triggerPoint(self, curve, point_index, i):
+        """
+        Method use to trigger the point
+
+        :param curve: is the reference to the current curve
+        :param point_index: index of picked point
+        :param i: the index of current curve
+        """
+
+        # Update itemas to make sure the older point is not still trigger
+        curve.updateItems()
+        scatterPoint = self.scatter[i].points()[point_index[0]]
+        scatterPoint.setSize(20)
+
+    def detach(self):
+        """
+        Removing the :class:`pyqtgraph.InfiniteLine` and
+        :class:`pyqtgraph.TextItem` from the plot
+        """
+        self.reset_custom_style()
+        self.plot.scene().removeItem(self.v_line)
+        for cloud in self.informationClouds.values():
+            self.plot.scene().removeItem(cloud)
+
+    @staticmethod
+    def getPointIndex(np_curve_data, mouse_point):
```

Comment:
This method should accept a parameter to set the tolerance (i.e. to be passed to `point_picker.checkPoint` as `roi`). The optimal value of that parameter should be calculated by `DataInspectorModel` taking into account the current scale. More specifically, IMHO the tolerance should be calculated considering the symbol size (e.g. 20 pixels) and translating it to a range in x axis coordinates)

Reply:
This is a good idea to optimize the algorytm finding the point on the curves.

I implemented very simple ROI calculation based on the pixel size and symbol size
(calculation relies on the minimum and maximum value in the X axis).